components/CostsCard.vue

```diff
@@ -0,0 +1,290 @@
+
+// The first of three candidate styles that will be evaluated for how much space they can provide for the pie.
+const topRightStyle = computed(() => {
+  // What is the maximum size the pie can be if it is in the top right corner?
+  const availableWidthInTopRightCorner = containers.cardBody.width.value - Math.max(containers.gdp.width.value, containers.usd.width.value);
+  const availableHeightInTopRightCorner = containers.cardBody.height.value;
+  const maxSizeInTopRight = Math.min(availableWidthInTopRightCorner, availableHeightInTopRightCorner);
+  let rightPosition = 0;
+  if (maxSizeInTopRight < availableWidthInTopRightCorner) {
+    // If spare horizontal space, calculate a value for the 'right' CSS property for centering.
+    rightPosition = (availableWidthInTopRightCorner - maxSizeInTopRight) / 2;
+  }
+  return {
+    height: maxSizeInTopRight,
+    right: rightPosition,
+    top: 0,
+  };
+});
+
+// The second of three candidate styles that will be evaluated for how much space they can provide for the pie.
+const belowUsdTotalStyle = computed(() => {
+  // What is the maximum size the pie can be if it is entirely below the USD total cost container?
+  const availableHeightBelowUsdTotalCost = containers.cardBody.height.value - containers.totals.height.value;
+  const maxSizeBelowUsd = Math.min(containers.cardBody.width.value, availableHeightBelowUsdTotalCost);
+  let rightPosition = 0;
+  if (maxSizeBelowUsd < containers.cardBody.width.value) {
+    // If spare horizontal space, calculate a value for the 'right' CSS property for centering.
+    rightPosition = (containers.cardBody.width.value - maxSizeBelowUsd) / 2;
+  };
+  return {
+    height: maxSizeBelowUsd,
+    right: rightPosition,
+    bottom: 0,
+  };
+});
+
+// The third of three candidate styles that will be evaluated for how much space they can provide for the pie.
+const bottomRightStyle = computed(() => {
+  // What is the maximum size the pie can be if it is in the bottom right, and able to expand into the cleft between the GDP total cost and USD total cost?
+  const availableWidthInBottomRightCorner = containers.cardBody.width.value - containers.usd.width.value;
+  const availableHeightInBottomRightCorner = containers.cardBody.height.value - (containers.totals.height.value - containers.usd.height.value);
+  const maxSizeInBottomRight = Math.min(availableWidthInBottomRightCorner, availableHeightInBottomRightCorner);
+  let rightPosition = 0;
+  if (maxSizeInBottomRight < availableWidthInBottomRightCorner) {
+    // If spare horizontal space, calculate a value for the 'right' CSS property for centering.
+    rightPosition = (containers.cardBody.width.value - maxSizeInBottomRight) - containers.usd.width.value;
+  }
+  return {
+    height: maxSizeInBottomRight,
+    right: rightPosition,
+    bottom: 0,
+  };
+});
+
+const styleWithLargestPie = computed(() => [topRightStyle.value, belowUsdTotalStyle.value, bottomRightStyle.value].sort((a, b) => b.height - a.height)[0]);
```

Comment:
Or could be something like
```suggestion
const styleWithLargestPie = computed(() => Math.max(...[topRightStyle.value, belowUsdTotalStyle.value, bottomRightStyle.value].map((s) => s.height));
```
..but just because I don't love reading sort methods!

Reply:
Doesn't that return a number (a height) rather than a style? Or is it some fancy syntax I've not seen